tests/imports.py

```diff
@@ -23,69 +23,456 @@ def _strip_magic_module_attributes(ls: list) -> list:
-            "ecephys",
-            "icephys",
-            "ophys",
-            "text",
-            # Exposed attributes
-            "interface_list",
-            "interfaces_by_category",
-        ] + interface_name_list
-
-        assert sorted(current_structure) == sorted(expected_structure)
+def test_top_level():
+    import neuroconv
+
+    current_structure = _strip_magic_module_attributes(ls=neuroconv.__dict__)
+    expected_structure = [
+        # Sub-modules
+        "basedatainterface",
+        "basetemporalalignmentinterface",
+        "baseextractorinterface",
+        "nwbconverter",
+        "tools",  # Attached to namespace by NWBConverter import
+        "utils",  # Attached to namespace by NWBconverter import
+        # Exposed attributes
+        "NWBConverter",
+        "ConverterPipe",
+        "BaseDataInterface",
+        "BaseTemporalAlignmentInterface",
+        "BaseExtractorInterface",
+        "run_conversion_from_yaml",
+    ]
+    assert sorted(current_structure) == sorted(expected_structure)
+
+
+def test_tools():
+    """Python dir() calls (and __dict__ as well) update dynamically based on global imports."""
+    from neuroconv import tools
+
+    current_structure = _strip_magic_module_attributes(ls=tools.__dict__)
+    expected_structure = [
+        # Sub-Packages
+        "yaml_conversion_specification",  # Attached to namespace  by top __init__ call of NWBConverter
+        # Sub-modules
+        "importing",  # Attached to namespace by importing get_package
+        "nwb_helpers",  # Attached to namespace by top __init__ call of NWBConverter
+        "path_expansion",
+        # Functions and classes imported on the __init__
+        "get_package",
+        "processes",
+        "deploy_process",
+        "LocalPathExpander",
+        "get_module",
+        "hdmf",
+    ]
+    assert sorted(current_structure) == sorted(expected_structure)
+
+
+def test_datainterfaces():
+    from neuroconv import datainterfaces
+
+    current_structure = _strip_magic_module_attributes(ls=datainterfaces.__dict__)
+
+    from neuroconv.datainterfaces import interface_list
+
+    interface_name_list = [interface.__name__ for interface in interface_list]
+    expected_structure = [
+        # Sub-modules
+        "behavior",
+        "ecephys",
+        "icephys",
+        "ophys",
+        "text",
+        # Exposed attributes
+        "interface_list",
+        "interfaces_by_category",
+    ] + interface_name_list
+
+    assert sorted(current_structure) == sorted(expected_structure)
+
+
+def test_guide_attributes():
+    from neuroconv.datainterfaces import interface_list
+
+    guide_attribute_names = ["display_name", "keywords", "associated_suffixes", "info"]
+    guide_attributes_per_interface = {
+        interface.__name__: {attribute: getattr(interface, attribute) for attribute in guide_attribute_names}
+        for interface in interface_list
+    }
+
+    assert guide_attributes_per_interface == {
```

Comment:
It's a bit structure to have in the test; I can maybe offload it to an external JSON file

But it sure it nice to be able to see all the results aggregated together

Some comments below about standardizing text

Reply:
JSON doesn't preserve the typing in this case

Reply:
Could go with `pickle` I suppose but that defeats the purpose of human readability